apollo-gradle-plugin/src/main/kotlin/com/apollographql/apollo/gradle/internal/DefaultService.kt

```diff
@@ -20,14 +21,22 @@ import javax.inject.Inject
 abstract class DefaultService @Inject constructor(val objects: ObjectFactory, override val name: String)
   : Service {
 
+  init {
+    // see https://github.com/gradle/gradle/issues/7485
+    // TODO replace with `convention(null)` when we can target Gradle 6.2
```

Comment:
Can we not target 6.2 now?

Reply:
This was filed not so long ago: https://github.com/apollographql/apollo-android/issues/2536

Overall, I don't think this alone justifies bumping the Gradle version. We could do something like this:

```
if (gradleVersion >= 6.2) {
   customTypeMapping.convention(null)
} else {
   customTypeMapping.set(null as Map<String, String>?)
}
```

This would make it possible for 6.2+ users to do something like

```
apollo {
  customTypeMapping.put("Date", "java.util.Date")
}
```

(in addition to the always working)

```
apollo {
  customTypeMapping.set(mapOf("Date" to "java.util.Date"))
}
```

Reply:
Hmm, I was thinking because of 3.x but you're right. If we don't need it, we shouldn't.

Your suggestion also looks like a good improvement. I didn't know that. It is up to you.

Reply:
I added it, it feels more natural to have proper `.put` support